### PR TITLE
unxip 1.1.3 (new formula)

### PR DESCRIPTION
**Pull request by timsutton**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/unxip.rb
Comment (timsutton) [inline Formula/unxip.rb:22]: Since this needs a whole Xcode .xip in order to test functionality, this is about the best we can do today as far as a `test`.

File: Formula/unxip.rb
Comment (timsutton) [inline Formula/unxip.rb:13]: Since `unxip` uses APIs that aren't available on Linux, it won't make sense for it to be installable on Linux. Can I just add a separate line around here that says:

```
  depends_on :macos
```
? Or is there a preferred way to do this?

Comment (chenrui333) [reply Formula/unxip.rb:13]: yeah, separate line `depends_on :macos`

File: Formula/unxip.rb
Comment (chenrui333) [inline Formula/unxip.rb:12]: this is not needed
```suggestion
```

File: Formula/unxip.rb
Comment (chenrui333) [inline Formula/unxip.rb:17]: why not use this kind of standardized build line

```suggestion
    system "swift", "build", "--disable-sandbox", "-c", "release"
```

File: Formula/unxip.rb
Comment (timsutton) [reply Formula/unxip.rb:9]: I think that when I added this line, CI created a [tests job for Big Sur](https://github.com/Homebrew/homebrew-core/actions/runs/3415668383/jobs/5685109396) when it hadn't before when I had only `depends_on macos: :monterey`. Is that possible?

File: Formula/unxip.rb
Comment (timsutton) [reply Formula/unxip.rb:13]: I got these build options via the maintainer's docs, but he'd also given me guidance to avoid using SPM here so that it wouldn't require an Xcode to build from source (that just the Command Line Tools will suffice).

**Comment by chenrui333** [comment]
it dos not build successfully?

**Comment by timsutton** [comment]
> it dos not build successfully?

From what I can see it's fairly complicated to build it on Big Sur because it uses newer Swift concurrency libs that would need to be backdeployed along with the Swift runtime: https://github.com/saagarjha/unxip/issues/20#issuecomment-1148390867

I figured that at least as a start it might be easiest to just require Monterey. I think when I had just `depends_on macos: :monterey` the CI excluded Big Sur runs, but I also needed to add `depends_on :macos` in order to exclude Linux.

**Comment by Bo98** [comment]
We don't support double requirements like that but we probably should: https://github.com/Homebrew/brew/pull/14122

**Comment by timsutton** [comment]
Thanks! Would supporting this in the formula (and CI) need then to wait for a new Homebrew release including https://github.com/Homebrew/brew/pull/14122 which was just merged? 💯

**Comment by Bo98** [comment]
Ideally yeah, though CI at least will pick it up right away so I'll rerun that.

**Comment by timsutton** [comment]
I saw the new brew release was cut 👍 

Do you think this formula is otherwise in good shape to merge?

**Comment by timsutton** [comment]
@chenrui333 Just another friendly bump on this PR 🙏

**Comment by timsutton** [comment]
@iMichka @carlocab Wondered if either of you would be able to give this new formula PR a review? I'm not sure if there's another preferred approach to solicit other reviewers 🙏

**Comment by timsutton** [comment]
@SMillerDev Would you have a moment to take a look at reviewing/merging this new formula? It feels like it's in good shape, but I'm just having trouble getting some  👀  on this one. Thanks!

File: Formula/unxip.rb
Comment (SMillerDev) [reply Formula/unxip.rb:13]: Rarely anyone using Homebrew builds from source since there are prebuilt binaries available

File: Formula/unxip.rb
Comment (SMillerDev) [reply Formula/unxip.rb:18]: How about trying to open a txt file and seeing if it fails correctly?

**Review by SMillerDev — APPROVED** [review]
Thanks @timsutton  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

File: Formula/unxip.rb
Comment (timsutton) [reply Formula/unxip.rb]: Thanks for the suggestion! It looks like the best I could do for now as a test, is to create a dummy file that's close-ish to a .xip, attempt to expand to a nonexistent directory and test the exit status and error message.

`xip` built-in can actually create xip archives, but they require a signing identity, and only Apple identities are trusted on the system anyway.

File: Formula/unxip.rb
Comment (timsutton) [reply Formula/unxip.rb]: Ok, I've switched to this standardized build line instead and added `depends_on xcode: :build` to the formula.

**Pull request closed without merging**
